packages/hdwallet-provider/src/index.ts

```diff
@@ -131,6 +108,21 @@ class HDWalletProvider {
     const tmpAccounts = this.addresses;
     const tmpWallets = this.wallets;
 
+    // if user supplied the chain id, use that - otherwise fetch it
+    if (chainId || (chainSettings && chainSettings.chainId)) {
+      this.chainId = chainId || chainSettings.chainId;
```

Comment:
`chainId` could be `0`, so you'll need to check for null/undefined explicitly

Reply:
Ah yes, I didn't think of that.